src/go/rdctl/cmd/reset.go

```diff
@@ -0,0 +1,164 @@
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/factoryreset"
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/paths"
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/shutdown"
+)
+
+var (
+	vmReset      bool
+	k8sReset     bool
+	cacheReset   bool
+	factoryReset bool
+)
+
+var resetCmd = &cobra.Command{
+	Use:   "reset",
+	Short: "Reset Rancher Desktop",
+	Long: `Reset Rancher Desktop with various options:
+* --vm: Delete VM and create a new one with current settings
+* --k8s: Delete deployed Kubernetes workloads
+* --cache: Delete cached Kubernetes images
+* --factory: Delete VM and show first-run dialog on next start
+
+Options can be combined. Some combinations are redundant:
+* --factory includes --vm and --k8s (but not --cache)
+* --vm includes --k8s
+
+At least one option must be specified.`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if err := cobra.NoArgs(cmd, args); err != nil {
+			return err
+		}
+		cmd.SilenceUsage = true
+
+		// Check if any options are specified
+		if !vmReset && !k8sReset && !cacheReset && !factoryReset {
+			return fmt.Errorf("no reset options specified. Use --help to see available options")
+		}
+
+		// Handle factory reset (includes VM and K8s reset)
+		if factoryReset {
+			return performFactoryReset(cmd.Context(), cacheReset)
+		}
+
+		// Handle VM reset (includes K8s reset)
+		if vmReset {
+			if err := performVMReset(cmd.Context()); err != nil {
+				return err
+			}
+			// VM reset includes K8s, so cache is handled separately below
+		} else if k8sReset {
+			// Handle K8s-only reset
+			if err := performK8sReset(cmd.Context()); err != nil {
+				return err
+			}
+		}
+
+		// Handle cache reset if requested (and not already handled by factory reset)
+		if cacheReset {
+			return factoryreset.DeleteCacheData()
+		}
+
+		return nil
+	},
+}
+
+// performFactoryReset performs a factory reset with the given context and cache removal option
+func performFactoryReset(ctx context.Context, removeCache bool) error {
+	pathsCfg, err := paths.GetPaths()
+	if err != nil {
+		return fmt.Errorf("failed to get paths: %w", err)
+	}
+	commonShutdownSettings.WaitForShutdown = false
+	_, err = doShutdown(ctx, &commonShutdownSettings, shutdown.FactoryReset)
+	if err != nil {
+		return err
+	}
+	return factoryreset.DeleteData(ctx, pathsCfg, removeCache)
+}
+
+// ResetPayload defines the payload structure for reset requests
+type ResetPayload struct {
+	Mode string `json:"mode"`
+}
+
+// performVMReset performs a VM reset (includes K8s reset)
+func performVMReset(ctx context.Context) error {
+	result, err := doReset(ctx, "wipe")
+	if err != nil {
+		return err
+	}
+	fmt.Println(string(result))
+	return nil
+}
+
+// performK8sReset performs a K8s-only reset
+func performK8sReset(ctx context.Context) error {
+	result, err := doReset(ctx, "fast")
+	if err != nil {
+		return err
+	}
+	fmt.Println(string(result))
+	return nil
+}
```

Comment:
These are similar enough that maybe we should just passing the type (`wipe`/`fast`) into a shared function?  But then there's no point in having separate functions; that would mean the caller becomes:

```go
if vmReset || k8sReset {
  resetType := "wipe"
  if !vmReset {
     resetType = "fast"
  }
  result, err := doReset(cmd.Context(), resetType)
  if err != nil {
    return err
  }
  fmt.Println(string(result))
}
```